File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        k = k % nums.length; 
        int temp[] = new int[k];
        
        
        for (int i = 0; i < k; i++) {
            temp[i] = nums[nums.length - k + i];
        }
        
        
        for (int i = nums.length - 1; i >= k; i--) {
            nums[i] = nums[i - k];
        }
        
        
        for (int i = 0; i < k; i++) {
            nums[i] = temp[i];
        }
    }
}

File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:

    void reverse(vector<int> &arr,int s,int e){
        while(s<e){
            swap(arr[s],arr[e]);
            s++;
            e--;
        }
    }
    void printArray(const vector<int>& arr) {
    for (int num : arr) {
        cout << num << " ";
    }
    cout << endl;
}
    void rotate(vector<int>& nums, int k) {
            
         

        int n=nums.size();
        k = k % n;
        reverse(nums,0,n-1);
        reverse(nums,0,k-1);
        reverse(nums,k,n-1);

        printArray(nums);
         }
    

    
};